docs/how-to/create-epics-ioc-with-ca-and-pva.md

```diff
@@ -0,0 +1,951 @@
+        response = await self._connection.send_query("ERRORS?\r\n")
+        error_count = int(response.strip())
+        await self.error_count.update(error_count)
+```
+
+## Converting an Existing EPICS IOC to FastCS
+
+### Step 1: Analyze Your Current Implementation
+
+Identify the key components:
+
+1. **Database files** (`.db`, `.template`): Define your records
+2. **Device support**: C/C++ code or asynDriver configuration
+3. **Protocol files**: StreamDevice protocols or custom communication
+4. **Startup scripts**: IOC initialization
+
+### Step 2: Map EPICS Records to FastCS Attributes
+
+| EPICS Record Type | FastCS Equivalent | Notes |
+|------------------|-------------------|-------|
+| `ai` (Analog Input) | `AttrR(Float())` | Read-only numeric |
+| `ao` (Analog Output) | `AttrRW(Float())` | Read-write numeric, creates `_RBV` |
+| `bi` (Binary Input) | `AttrR(Bool())` | Read-only boolean |
+| `bo` (Binary Output) | `AttrRW(Bool())` | Read-write boolean |
+| `stringin` | `AttrR(String())` | Read-only string |
+| `stringout` | `AttrRW(String())` | Read-write string |
+| `longin` | `AttrR(Int())` | Read-only integer |
+| `longout` | `AttrRW(Int())` | Read-write integer |
+| `mbbi` (Multi-bit Binary Input) | `AttrR(Enum(...))` | Read-only enumeration |
+| `mbbo` (Multi-bit Binary Output) | `AttrRW(Enum(...))` | Read-write enumeration |
+| `waveform` | `AttrR(Waveform(...))` or `AttrRW(Waveform(...))` | Arrays |
+| `calc`, `calcout` | Use Python logic in controller | Calculations in code |
+
+### Step 3: Example Conversion
+
+**Original EPICS Database (`.db` file):**
+
+```
+record(ai, "$(P):TEMPERATURE") {
+    field(DTYP, "asynFloat64")
+    field(INP, "@asyn($(PORT)) TEMP")
+    field(SCAN, "1 second")
+    field(PREC, "2")
+    field(EGU, "°C")
+}
+
+record(ao, "$(P):SETPOINT") {
+    field(DTYP, "asynFloat64")
+    field(OUT, "@asyn($(PORT)) SETP")
+    field(PREC, "2")
+    field(EGU, "°C")
+}
+
+record(stringin, "$(P):ID") {
+    field(DTYP, "asynOctetRead")
+    field(INP, "@asyn($(PORT)) ID")
+}
+```
+
+**Equivalent FastCS Controller:**
+
+```python
+from fastcs.attributes import AttrR, AttrRW
+from fastcs.datatypes import Float, String
+from fastcs.controllers import Controller
+
+class TemperatureController(Controller):
+    """Temperature controller - replaces EPICS database."""
+
+    # Read-only temperature (ai record)
+    temperature: AttrR = AttrR(
+        Float(),
+        description="Current temperature in °C",
+        io_ref=DeviceAttributeIORef(command="TEMP", update_period=1.0)
+    )
+
+    # Read-write setpoint (ao record)
+    setpoint: AttrRW = AttrRW(
+        Float(),
+        description="Temperature setpoint in °C",
+        io_ref=DeviceAttributeIORef(command="SETP", update_period=1.0)
+    )
+
+    # Read-only ID (stringin record)
+    device_id: AttrR = AttrR(
+        String(),
+        description="Device identifier",
+        io_ref=DeviceAttributeIORef(command="ID", update_period=10.0)
+    )
+```
```

Comment:
This example isn't very useful as it oversimplifies what the EPICS db would really look like